services/settings/pkg/service/v0/servicedecorator.go

```diff
@@ -82,3 +94,93 @@ func (s *defaultLanguageDecorator) withDefaultLanguageSetting(accountUUID string
 		},
 	}
 }
+
+func (s *defaultLanguageDecorator) withDefaultProfileValue(ctx context.Context, accountUUID string, settingId string) *settingsmsg.ValueWithIdentifier {
+	if settingId == defaults.SettingUUIDProfileLanguage {
+		return s.withDefaultLanguageSetting(accountUUID)
+	}
+	res := s.withDefaultProfileValueList(ctx, accountUUID, map[string]*settingsmsg.ValueWithIdentifier{settingId: nil})
+	if v, ok := res[settingId]; ok {
+		return v
+	}
+	return nil
+}
+
+func (s *defaultLanguageDecorator) withDefaultProfileValueList(ctx context.Context,
+	accountUUID string, requested map[string]*settingsmsg.ValueWithIdentifier) map[string]*settingsmsg.ValueWithIdentifier {
+
+	// we use the default profile bundle instead of s.GetBundle(ctx, req, resp)
+	bundle := defaults.GenerateDefaultProfileBundle()
+
+	for _, setting := range bundle.GetSettings() {
+		if v, ok := requested[setting.GetId()]; ok && v == nil {
```

Comment:
Can we reverse the logic here? Too much intendation :)
```suggestion
		if v, ok := requested[setting.GetId()]; !ok || v != nil {
		    continue
		}
```